src/views/AddItem.jsx

```diff
@@ -1,3 +1,93 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { addItem } from '../api/firebase';
+import { useState } from 'react';
+
+export function AddItem({ listId }) {
+	// declare itemName, daysUntilNextPurchase, and radioSelect
+	// into separate state variables
+	const [itemName, setItemName] = useState('');
+
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(null);
+
+	const [radioSelect, setRadioSelect] = useState('soon');
+
+	// Store values inside itemData and sent over to database
+	// otherwise log an error if request fails
+	const onFormSubmit = (event) => {
+		event.preventDefault();
+		const itemData = {
+			itemName,
+			daysUntilNextPurchase,
+		};
+		try {
+			addItem(listId, itemData);
+			alert(`${itemData.itemName} was added to your list`);
+		} catch (error) {
+			console.log(error);
+		}
+	};
+
+	// Get user input and set itemName
+	const handleItemName = (event) => {
+		setItemName(event.target.value);
+	};
+
+	// Get radio selection and set daysUntilNextPurchase
+	const handlePurchaseDate = (event) => {
+		let date = null;
+		if (event.target.value === 'soon') {
+			date = 7;
+		} else if (event.target.value === 'kind-of-soon') {
+			date = 14;
+		} else if (event.target.value === 'not-soon') {
+			date = 30;
+		}
+		setDaysUntilNextPurchase(date);
+		setRadioSelect(event.target.value);
+	};
+
+	return (
+		<form onSubmit={onFormSubmit}>
+			<div>
+				<label htmlFor="item">Item Name:</label>
+			</div>
+			<input type="text" id="item" name="item" onChange={handleItemName} />
+			<div>How soon will you buy this again?</div>
+			<fieldset>
+				<div>
+					<input
+						checked={radioSelect === 'soon'}
```

Comment:
I'm not totally convinced this `checked` attribute is needed on every `input`. You can add it to one of the inputs to indicate that it should be the default option selected. So for `kind-of-soon` and `not-soon` it shouldn't ever be the default since you are keeping track of this state above. Along those same lines, not sure tracking that state (`radioSelect`) is necessary either.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#checking_boxes_by_default

Reply:
Good catch.  I did not realize we had all three checked.

Reply:
@djtaylor8 Thanks DJ! I actually originally had `checked` on only one input which is the "Soon" input. None of the other ones had `checked` on them. However, this was happening: 

![checked](https://user-images.githubusercontent.com/102492480/213824994-8cd2726a-9bb2-4711-8d57-2145f629b3ad.gif)

You'll notice when I click on the other radio buttons it jumps back to "Soon". So I opted to do it the way I did it here in our code. Why do you think this was happening to begin with?

Reply:
Ah good catch! Off the top of my head I'm not totally sure honestly. Maybe do some Googling to see if there are any leads there?

Reply:
@djtaylor8 Found a fix (hopefully the right fix. Tested several times with no errors at least). Merely changed `checked` to `defaultChecked` on the first `input`.